frida_mode/README.md

```diff
@@ -0,0 +1,34 @@
+# FRIDA MODE
+The purpose of FRIDA mode is to provide an alternative binary only fuzzer for AFL
+just like that provided by QEMU mode. The intention is to provide a very similar
+user experience, right down to the options provided through environment variables.
+
+Additionally, the intention is to be able to make a direct performance comparison
+between the two approaches. Hopefully, we should also be able to leverage the same
+approaches for adding features which QEMU uses, possibly even sharing code.
+
+## Limitations
+The current focus is on x64 support for Intel. Although parts may be architecturally
+dependent, the approach itself should remain architecture agnostic.
+
+## Usage
+FRIDA mode requires some small modifications to the afl-fuzz and similar tools in
+AFLplusplus. The intention is that it behaves identically to QEMU, but uses the 'O'
```

Comment:
I would argue against adding a new switch, just calling the `frida-trace` binary is fine imho. This leaves greater flexibility to the user. Also `O` seems random.

Reply:
IMHO this is ok. Like -W wine, -Q qemu and -U unicorn. -F would be better but that it is taken, same as -f. if there is something more fitting than -O - sure.

Reply:
This can just use -U and then `frida-trace`. It's something hidden from the user that has hardly any benefit imho.
Even -U I would probably scrap to be honest, and make SKIP_BINCHECK default

Reply:
we can not set skip bincheck off default. this is what detects persistent mode and deferred forkserver.
and removing -U would just add complexity to using it.

Reply:
I'm not sure I follow? `afl-frida-trace` is a shared object rather than an EXE. I don't think I can make it an executable as and executable cannot dlopen another executable (even if it is PIE which effectively makes it a DSO anyway) as the dynamic loader complains.

f & s were the obvious choices (as it is supported by FRIDAs stalker engine), but they were taken. Kind of running out of alphabet! So I went with O in homage to [Ole](https://github.com/oleavr) (the author). Happy to change though if you have a preference?

Reply:
> we can not set skip bincheck off default. this is what detects persistent mode and deferred forkserver.
> and removing -U would just add complexity to using it.

Persistent mode and deferred forkserver could be agreed upon done during forkserver handshake btw ;)